### PR TITLE
Issue983 axis qcplot edits vincent

### PR DESCRIPTION
**Pull request by vincentvanhees**
@jhmigueles In this PR I propose a couple of edits to PR #985

- Renaming of object names to improve readability
- Fixed day numbering, which missed the first incomplete calendar day. This is now accounted for.
- Your change to argument `mar` caused y axis to overlap for devices with temperature data.
- I replaced d1, d2, d3, etc by a single axis label "recording day" and then all the numbers at the axis.
- I changed x-axis font to 1 to be in line with y-axis, similarly the hash area for study protocol, and accelerometer signal has thinner lines now. I do this as pdf files tend to be smaller with thinner lines, which becomes an issue if the user creates thousands of plots like these. Also, I am trying to avoid inconsistent line width when there is no functional value of the differences.
- minor edits to legend to clarify that by 'study protocol' we refer to the masking of time. I had to keep it short as there is limited space.
- Replaced repeated calculation of day length, which by the way also was problematic as you were calculating this on a vector with possible length of one if recording has only one noon or midnight. Further, this number would be incorrect if there is a day in the recording with 25 hours.
- UPDATE: Now horizontal axis simplified as vertical grid already clarifies when noon occurs, so no need for ticks

#### Example 1:
image

#### Example 2:
image

Checklist before merging:

- [ ] Existing tests still work (check by running the test suite, e.g. from RStudio).
- [ ] Added tests (if you added functionality) or fixed existing test (if you fixed a bug).
- [ ] Updated or expanded the documentation.
- [ ] Updated release notes in `inst/NEWS.Rd` with a user-readable summary. Please, include references to relevant issues or PR discussions.
- [ ] Added your name to the contributors lists in the `DESCRIPTION` file, if you think you made a significant contribution.

### CODE REVIEW

**Comment by vincentvanhees** [comment]
Actually, I think the double axis does not serve any purpose because the vertical lines already clarify when noon and midnight is. I will now make an extra commit to remove the double axis such that we only have one axis to indicate the midnights

**Comment by vincentvanhees** [comment]
I just made a few minor edits. The grid lines now cover all plots and the legend is more clearly positioned on top. To illustrate how it looks like with all masking types I simulated some clipping in the example I show above, which now looks:

image

File: R/g.plot.R [hunk 1]
```diff
@@ -176,50 +179,45 @@ g.plot = function(IMP, M, I, durplot) {
...
     }
   }
-  plot_nonwear = function(timeline, M, durplot, ticks) {
+  plot_nonwear = function(timeline, M, durplot, ticks_12hours) {
     plot(timeline, M$metalong$nonwearscore, type = "s",
          xlab = "", ylab = "Non-wear score", axes = F,
          lwd = 0.1, xlim = c(0,durplot), ylim = c(0, 3), cex.lab = 0.8)
-    axis(side = 2,at = c(0, 1, 2, 3))
-    # axis(side = 1,at = ticks, labels = 1:length(ticks))
+    axis(side = 2,at = c(0, 1, 2, 3), cex.axis = 0.8)
```
Comment (jhmigueles) [inline R/g.plot.R:197]: Maybe add `las = 1` in the axis definition so that y axis is consistent in the lower and upper plots.

File: R/g.plot.R [hunk 2]
```diff
@@ -176,50 +179,45 @@ g.plot = function(IMP, M, I, durplot) {
...
     
     par(fig = c(0, 1, 0.60, 0.95), new = T)
     plot(timeline, M$metalong$temperaturemean[1:MEND], type = "l",
          xlab = "", ylab = "Temp. (C)", axes = F, lwd = 0.1,
          xlim = c(0, durplot), ylim = c(20,35), cex.lab = 0.8)
     abline(h = 20, col = "black", lwd = 1, lty = 2)
     abline(h = 35, col = "black", lwd = 1, lty = 2)
-    axis(side = 2,at = c(20,35))
-    # axis(side = 1,at = ticks, labels = 1:length(ticks))
+    axis(side = 2,at = c(20,35), cex.axis = 0.8)
```
Comment (jhmigueles) [inline R/g.plot.R:214]: Also add `las = 1` here.

**Review by jhmigueles — APPROVED** [review]
Looks great! Just a couple of minor comments to make the y axis labels always perpendicular to the axis in all plots (ACC, temperature, and nonwear score). No further comments beyond that. Thanks!

**Pull request merged**